### ISSUE TITLE
Client setting are ignored for CNAME matching

### ISSUE THREAD

**Issue by davidbdeath** [AdguardTeam/AdGuardHome#1274]
Client setting now working v0.100.2. 
Tried MAC and IP identifier. 
Using adguard DHCP. 
Rasberry Pi host, 
shield tv client.
Static IP

**Comment by szolin** [comment]
what doesn't work exactly?
what's your configuration, what you are expecting and what's the current behaviour?

**Comment by davidbdeath** [comment]
The client settings are not working. I was expecting the config for the shield to work and not be ignored. I disabled filtering for the shield so my wife could watch the CW. Adguard ignored the client settings i specified for the shield. My setup is what I listed above.

**Comment by szolin** [comment]
Can't reproduce.  Please give me more details.  What's the client ID?  What's your DNS request?  What's the DNS response?

**Comment by SmakyTaky** [comment]
I am experiencing the same issue. I have setup a few clients to manage exceptional settings and none of them are being applied. e.g. one client based on IP address is set to override all blocking... however browsing from that client still get blocked unless AdGuard Home is disabled altogehter.

**Comment by NetE1** [comment]
> I noticed on the Query Log page that all records show client IP as 0.0.0.0 thus not recognizing the client the request is comming from and thus applying global settings to the query. Resulting in being blocked.

It must be your end problem because seems agh process requests but can't identify each of them. If you added a client that also appeared on top clients table then query log entries will display correctly and you'll not experience any custom settings issue. by the way, are those clients connect directly to server?
https://github.com/AdguardTeam/AdGuardHome/issues/1220

> I was expecting the config for the shield to work and not be ignored. I disabled filtering for the shield so my wife could watch the CW.

It'll be better if you can show us related DNS request and response records as @szolin said :)

**Comment by davidbdeath** [comment]
My clients show an IP address on the query log screen.

example would be ssl.google-analytics.com with a response of blocked by CNAME or IP A: 172.217.9.72 (ttl=29)

**Comment by ArchiveBase** [comment]
It's a bug @szolin 

example to reproduce
- block akamai technologies or ex: ||akadns.net^
- add a client with global settings unchecked
- ping time.windows.com (or run windows time sync)

**Comment by NullEnt1ty** [comment]
Not sure if this is the case here, but I noticed that the new feature "Match CNAME records against the blocklists" is ignoring client settings. I had a swift look over [the responsible commit](https://github.com/AdguardTeam/AdGuardHome/commit/07a9c3bd452e474d0de31cce6b321156e9672b23) and did not find any client settings related code.

![image](https://user-images.githubusercontent.com/966433/71325015-f2a1f600-24e6-11ea-9655-1a65dd530291.png)

You can see two clients in the screenshot above. The client _dominique-pc_ is exempted from any filtering while the other one is not. Both requests were filtered regardless of the client's settings.

I'm using AdGuard Home v0.100.6 (in Docker).

**Comment by ArchiveBase** [comment]
@NullEnt1ty that's what i mentioned above

**Comment by davidbdeath** [comment]
Another Issue i'm sure is related, i don't have 'use safe search' enabled but Adguard is forcing safe search.

**Comment by ameshkov** [comment]
> "Match CNAME records against the blocklists" is ignoring client settings

It's enabled by default, there's no option to turn it off.

**Comment by ameshkov** [comment]
> I noticed on the Query Log page that all records show client IP as 0.0.0.0 thus not recognizing

@SmakyTaky could you please record AdGuard Home logs for us? I don't see how it could happen.

1. [Configure](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file) AdGuard Home to collect logs:
    * Specify `log_file`
    * Set `verbose` to `True`
2. Restart AdGuard Home and reproduce the issue
3. Post the log file here.

**Comment by davidbdeath** [comment]
I am the network administrator and I have not enabled restricted mode.

**Comment by ameshkov** [comment]
@davidbdeath please try installing the beta version, the issue should've been fixed there: https://github.com/AdguardTeam/AdGuardHome#test-unstable-versions
